garden_ai/client.py

```diff
@@ -353,7 +359,41 @@ def register_metadata(self, garden: Garden, out_dir=None):
             with open(out_dir / f"{garden.uuid}.json", "w+") as f:
                 f.write(garden.json())
 
-    def get_local(self, uuid: UUID) -> JSON:
+    def put_local_garden(self, garden: Garden) -> None:
+        """Helper: write a record to 'local database' for a given Garden or Pipeline.
+
+        Overwrites any existing entry with the same ``uuid`` in
+        ``~/.garden/db/data.json``.
+
+        Parameters
+        ----------
+        garden : Garden
+            The object to json-serialize and write/update in the local database.
+            a TypeError will be raised if not a Garden.
+
+        """
+        if not isinstance(garden, Garden):
+            raise TypeError(f"Expected Garden object, got: {type(garden)}.")
+        data = {}
+        # read existing entries into memory, if any
+        if (LOCAL_STORAGE / "data.json").exists():
+            with open(LOCAL_STORAGE / "data.json", "r+") as f:
+                raw_data = f.read()
+                if raw_data:
+                    data = json.loads(raw_data)
+
+        # update data['gardens'], leaving data['pipelines'] etc unmodified
+        gardens = data.get("gardens", {})
+        key, val = str(garden.uuid), garden.json()
+        gardens[key] = json.loads(val)
+        data["gardens"] = gardens
+        contents = json.dumps(data)
+
+        with open(LOCAL_STORAGE / "data.json", "w+") as f:
```

Comment:
I think we're missing some opportunities to DRY our persistence operations. For example it looks like this block

```
 if (LOCAL_STORAGE / "data.json").exists():
            with open(LOCAL_STORAGE / "data.json", "r+") as f:
                raw_data = f.read()
                if raw_data:
                    data = json.loads(raw_data)
```

is present in put_local_garden and put_local_pipeline.